src/providers/krb5/krb5_child.c

```diff
@@ -1464,7 +1464,7 @@ static errno_t changepw_child(struct krb5_req *kr, bool prelim)
     kerr = krb5_get_init_creds_password(kr->ctx, kr->creds, kr->princ,
                                         password_or_responder(password),
                                         prompter, kr, 0,
-                                        SSSD_KRB5_CHANGEPW_PRINCIPAL,
+                                        discard_const(SSSD_KRB5_CHANGEPW_PRINCIPAL),
                                         kr->options);
```

Comment:
Such patch was rejected by simo approximately 3 years ago.
Because it was fixed in krb5.